domain-management/src/main/java/org/jboss/as/domain/management/security/FileKeyManagerService.java

```diff
@@ -117,45 +141,117 @@ public void setAlias(String alias) {
+                        if (!eventContext.isResourceServiceRestartAllowed()) {
+                            eventContext.reloadRequired();
+                        }
+                    }
+
+                    @Override
+                    public void pathEvent(Event event, PathEntry pathEntry) {
+                        // Service dependencies should trigger a stop and start.
+                    }
+                }, Event.REMOVED, Event.UPDATED);
+            }
+            Path path = Paths.get(file);
+            if (!Files.exists(path) && autoGenerateCertHostName != null) {
+                if(startup) {
+                    DomainManagementLogger.SECURITY_LOGGER.keystoreWillBeCreated(file, autoGenerateCertHostName);
+                    return null;
+                } else {
+                    generateFileKeyStore(path);
                 }
-            }, Event.REMOVED, Event.UPDATED);
-        }
+            }
 
-        keyStore = FileKeystore.newKeyStore(provider, file, keystorePassword, keyPassword, alias);
-        keyStore.load();
+            keyStore = FileKeystore.newKeyStore(provider, file, keystorePassword, keyPassword, alias);
+            keyStore.load();
 
-        super.start(context);
+            return keyStore.getKeyStore();
+        } catch (StartException e) {
+            throw new IllegalStateException(e);
+        }
     }
 
-    @Override
-    public void stop(StopContext context) {
-        super.stop(context);
-        keyStore = null;
+    private void generateFileKeyStore(Path path) {
+        try {
+            DomainManagementLogger.SECURITY_LOGGER.keystoreWillBeCreated(path.toString());
+            KeyPairGenerator keyGen = KeyPairGenerator.getInstance("RSA");
+            keyGen.initialize(2048, new SecureRandom());
+            KeyPair pair = keyGen.generateKeyPair();
+            X509Certificate cert = generateCertificate(pair);
+
+
+            KeyStore keyStore = KeyStore.getInstance(KeyStore.getDefaultType());
+            keyStore.load(null, keystorePassword);
+
+            //Generate self signed certificate
+            X509Certificate[] chain = new X509Certificate[1];
+            chain[0] = cert;
+            keyStore.setKeyEntry(alias, pair.getPrivate(), keyPassword, chain);
+            try (FileOutputStream stream = new FileOutputStream(path.toFile())) {
+                keyStore.store(stream, keystorePassword);
+            }
+
+        } catch (Exception e) {
+            throw DomainManagementLogger.SECURITY_LOGGER.failedToGenerateSelfSignedCertificate(e);
+        }
     }
 
-    @Override
-    protected KeyStore loadKeyStore() {
-        return keyStore.getKeyStore();
+    X509Certificate generateCertificate(KeyPair pair) throws Exception {
+        PrivateKey privkey = pair.getPrivate();
+        X509CertInfo info = new X509CertInfo();
+        Date from = new Date();
+        Date to = new Date(from.getTime() + (1000L * 60L * 60L * 24L * 365L * 10L));
```

Comment:
Maybe not so short they need to constantly keep re-generating it but 10 years is such a long time to put this into production and forget about it.

One thing that is 100% guaranteed is any features in this server that are for development only will go into production.